cli/status/status.go

```diff
@@ -1,57 +1,185 @@
 package status
 
 import (
+	_ "embed"
+	"fmt"
 	"os"
+	"strings"
 
+	"github.com/fatih/color"
 	"github.com/jedib0t/go-pretty/v6/table"
 	"github.com/jedib0t/go-pretty/v6/text"
+	"github.com/mitchellh/mapstructure"
 	"github.com/tarantool/tt/cli/connector"
 	"github.com/tarantool/tt/cli/process_utils"
 	"github.com/tarantool/tt/cli/running"
 )
 
+//go:embed lua/instance_state.lua
+var replicationInfoLuaScript string
+
 // StatusOpts contains options for tt status.
 type StatusOpts struct {
 	// Option for pretty-formatted table output.
-	Pretty bool
+	Pretty  bool
+	Details bool
+}
+
+type Alert struct {
+	Type    string `mapstructure:"type"`
+	Message string `mapstructure:"message"`
+}
+
+type ConfigInfo struct {
+	Status string  `mapstructure:"status"`
+	Alerts []Alert `mapstructure:"alerts"`
+}
+
+type Upstream struct {
+	Status  string `mapstructure:"status"`
+	Message string `mapstructure:"message"`
+}
+
+type ReplicationInfo struct {
+	Name     string    `mapstructure:"name"`
+	Upstream *Upstream `mapstructure:"upstream"`
+}
+
+type InstanceState struct {
+	ReplicationInfo []ReplicationInfo `mapstructure:"replication_info"`
+	ConfigInfo      ConfigInfo        `mapstructure:"config_info"`
+	ReadOnly        string            `mapstructure:"read_only"`
+	BoxStatus       string            `mapstructure:"box_status"`
 }
 
 // Status writes the status as a table.
 func Status(runningCtx running.RunningCtx, opts StatusOpts) error {
 	ts := table.NewWriter()
 	ts.SetOutputMirror(os.Stdout)
-	ts.AppendHeader(table.Row{"INSTANCE", "STATUS", "PID", "MODE"})
+	ts.AppendHeader(
+		table.Row{"INSTANCE", "STATUS", "PID", "MODE", "CONFIG", "BOX", "UPSTREAM"})
+
+	instances := map[string]map[string]interface{}{}
+	instancesAlerts := map[string][]string{}
+
+	PrintYellow := color.New(color.FgYellow).SprintFunc()
+	PrintRed := color.New(color.FgRed).SprintFunc()
 
 	for _, run := range runningCtx.Instances {
-		row := []interface{}{}
 		fullInstanceName := running.GetAppInstanceName(run)
 		procStatus := running.Status(&run)
 
-		row = append(row, fullInstanceName)
-		row = append(row, procStatus.ColorSprint(procStatus.Status))
+		if _, exists := instances[fullInstanceName]; !exists {
+			instances[fullInstanceName] = map[string]interface{}{
+				"STATUS": "",
+				"PID":    nil,
+				"MODE":   "",
+				"CFG":    "--",
+				"BOX":    "--",
+				"UPS":    "--",
+			}
+		}
+
+		instances[fullInstanceName]["STATUS"] = procStatus.ColorSprint(procStatus.Status)
```

Comment:
If status is `ERROR`, should we add something into the alert list?

Reply:
I think no. The `ERROR` status means that the process died and we can't get a connection to get more information.

Reply:
I've a similar status-and-details composition in `config:info()` and here if `status` is `check_errors`, then `alerts` have at least one diagnostic (with the `error` type).

So, it seems logical for me to have at least some 'process is died' in the detailed alerts in the case. The opposite (bad status, no alerts) seems less logical.

OTOH, all the information is on the screen anyway, so I don't insist here at all. Feel free to close this discussion with any your decision.